src/mjolnir/ferry_connections.cc

```diff
@@ -337,23 +353,24 @@ void ReclassifyFerryConnections(const std::string& ways_file,
           // Driveable in both directions - get an inbound path and an
           // outbound path.
           total_count += ShortestPath(node_itr.position(), end_node_idx, ways, way_nodes, edges,
-                                      nodes, true, rc);
+                                      nodes, true, remove_destonly);
           total_count += ShortestPath(node_itr.position(), end_node_idx, ways, way_nodes, edges,
-                                      nodes, false, rc);
+                                      nodes, false, remove_destonly);
         } else {
           // Check if oneway inbound to the ferry
           bool inbound =
               (edge.first.sourcenode_ == node_itr.position()) ? edge_rev_access : edge_fwd_access;
           total_count += ShortestPath(node_itr.position(), end_node_idx, ways, way_nodes, edges,
-                                      nodes, inbound, rc);
+                                      nodes, inbound, remove_destonly);
         }
         ferry_endpoint_count++;
 
         // Reclassify the first/start edge. Do this AFTER finding shortest path so
         // we do not immediately determine we hit the specified classification
         sequence<Edge>::iterator element = edges[edge.second];
         auto update_edge = *element;
-        update_edge.attributes.importance = rc;
+        update_edge.attributes.importance = kFerryUpClass;
+        update_edge.attributes.reclass_ferry = remove_destonly;
```

Comment:
#1905 was missing this: the connecting edge itself also needs its destonly removed if there was one.